app/models/data_file.rb

```diff
@@ -83,6 +83,27 @@ def as_indexed_json(options={})
           raw: {type: "string", index: "not_analyzed"}
         }
       end
+      indexes :project do
+        indexes :id, type: "string"
+        indexes :name, type: "string"
+      end
+
+      indexes :parent do
+        indexes :id, type: "string"
+        indexes :name, type: "string"
+      end
+
+      indexes :creator do
+        indexes :id, type: "string"
+        indexes :username, type: "string"
+        indexes :first_name, type: "string"
+        indexes :last_name, type: "string"
+        indexes :email, type: "string"
+      end
     end
   end
+
+  def creator
+    current_file_version.audits.find_by(action: "create").user
```

Comment:
Might want to put a guard clause around this in case current_file_version, .audits, or .find_by(action: "create") return nil.